saga/sagalogs/memory_test.go

```diff
@@ -0,0 +1,104 @@
+package sagalogs
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/twitter/scoot/saga"
+)
+
+func TestMemoryStartLogGetMessages(t *testing.T) {
+	slog := MakeInMemorySagaLog(0, 1)
+
+	sagaId := "s1"
+	jobData := []byte{0, 1, 2, 3, 4, 5}
+	slog.StartSaga(sagaId, jobData)
+
+	loggedMsgs := []saga.SagaMessage{
+		saga.MakeStartSagaMessage(sagaId, jobData),
+		saga.MakeStartTaskMessage(sagaId, "task1", []byte("run task 1")),
+		saga.MakeEndTaskMessage(sagaId, "task1", []byte("success")),
+		saga.MakeAbortSagaMessage(sagaId),
+		saga.MakeStartCompTaskMessage(sagaId, "task1", []byte("rollingback task1")),
+		saga.MakeEndCompTaskMessage(sagaId, "task1", []byte("finished rollingback task1")),
+		saga.MakeEndSagaMessage(sagaId),
+	}
+
+	for i, msg := range loggedMsgs {
+		// skip start saga message we already did it
+		if i == 0 {
+			continue
+		}
+
+		err := slog.LogMessage(msg)
+		if err != nil {
+			t.Fatalf("Unexpected error logging saga message: %s", err)
+		}
+	}
+
+	if !isSagaInActiveList(sagaId, slog) {
+		t.Fatalf("Expected Saga %s to be in active list", sagaId)
+	}
+
+	outMsgs, err := slog.GetMessages(sagaId)
+	if err != nil {
+		t.Fatalf("Unexpected error getting saga messages: %s", err)
+	}
+	if outMsgs == nil {
+		t.Fatalf("Unexpected nil result from GetMessages")
+	}
+	if len(outMsgs) != len(loggedMsgs) {
+		t.Fatalf("GetMessages length did not match loggedMsgs: %d/%d", len(outMsgs), len(loggedMsgs))
+	}
+
+	// we expect the messages to be returned in the same order they are logged
+	for j, outMsg := range outMsgs {
+		if !reflect.DeepEqual(loggedMsgs[j], outMsg) {
+			t.Errorf("Expected Logged Message and Returned Message to be Equal.  Expected %v, Actual %v",
+				loggedMsgs[j], outMsg)
+		}
+	}
+}
+
+func TestMemoryGC(t *testing.T) {
+	// set GC slow enough that it won't happen during the test
+	slog := MakeInMemorySagaLog(1*time.Hour, 1*time.Hour)
```

Comment:
try to have different values here to pick up errors with parameter ordering

Reply:
yeah - will add some better param tests